src/tests/fs_tests.rs

```diff
@@ -4036,4 +4037,136 @@ pub mod fs_tests {
+
+        let cage = interface::cagetable_getref(1);
+
+        let mut statdata = StatData::default();
+
+        // test whether an invalid fd results in a panic
+        let invalid_fd = 0x10000000; // some dummy high fd which should fail fd check
+
+        // here, we catch the panic spat out by fstat into an Err
+        let invalid_fd_panic = catch_unwind(AssertUnwindSafe(|| {
+            cage.fstat_syscall(invalid_fd, &mut statdata);
+        }));
+        assert!(invalid_fd_panic.is_err());
+
+        // test out whether an error is output for a non existent fd (1000)
+        // (ENOENT[-2])
+        let non_existent_fd = 1000;
+        assert_eq!(cage.fstat_syscall(non_existent_fd, &mut statdata), -2);
+
+        // setting up directory inode object '/tmp' for testing fstat_syscall with a
+        // directory
+        let dir_path = "/tmp"; // since setup already initializes tmp, assuming it is there
+        let dir_fd = cage.open_syscall(dir_path, O_RDONLY | O_DIRECTORY, S_IRWXA);
+        assert!(dir_fd > 0);
+        assert_eq!(cage.fstat_syscall(dir_fd, &mut statdata), 0);
+        assert_eq!(cage.close_syscall(dir_fd), 0);
+
+        // setting up generic inode object "/tmp/generic" for testing fstat_syscall with
+        // a generic file
+        let generic_path = "/tmp/generic";
+        let creat_fd = cage.creat_syscall(generic_path, S_IRWXA);
+        assert!(creat_fd > 0);
+        assert_eq!(cage.fstat_syscall(creat_fd, &mut statdata), 0);
+
+        // setting up character device inode object "/chardev" for testing fstat_syscall
+        // with a character device
+        let dev = makedev(&DevNo { major: 1, minor: 3 });
+        let chardev_path = "/chardev";
+        assert_eq!(
+            cage.mknod_syscall(chardev_path, S_IRWXA | S_IFCHR as u32, dev),
+            0
+        );
+        let chardev_fd = cage.open_syscall(chardev_path, O_RDONLY, S_IRWXA);
+        assert!(chardev_fd > 0);
+        assert_eq!(cage.fstat_syscall(chardev_fd, &mut statdata), 0);
+        assert_eq!(cage.close_syscall(chardev_fd), 0);
+
+        // setting up socket inode object with path "/socket.sock"  for testing
+        // fstat_syscall with a socket
+        let socketfile_path = "/socket.sock";
+        let socketfd = cage.socket_syscall(AF_UNIX, SOCK_STREAM, 0);
+        assert!(socketfd > 0);
+        let sockaddr = interface::new_sockaddr_unix(AF_UNIX as u16, socketfile_path.as_bytes());
+        let socket = interface::GenSockaddr::Unix(sockaddr);
+        assert_eq!(cage.bind_syscall(socketfd, &socket), 0);
+
+        // fstat_syscall test here
+        let socketfile_path_fd = cage.open_syscall(socketfile_path, O_RDONLY, S_IRWXA);
+        assert!(socketfile_path_fd > 0);
+        assert_eq!(cage.fstat_syscall(socketfile_path_fd, &mut statdata), 0);
+        assert_eq!(cage.close_syscall(socketfile_path_fd), 0);
+
+        // socket teardown
+        assert_eq!(cage.close_syscall(socketfd), 0);
```

Comment:
is the cleanup necessary here or does lindrustfinalize take care of that?

Reply:
so imo its always a good habit to teardown things once it is done being used since that makes sure that the resource is cleaned up. `lindrustfinalize` just deals with 'domain' socket inode, but im not sure what that means though, so i thought there was no harm handling this part myself